### PR TITLE
Notes List: Swifting UITableViewDatasource

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're reimplementing UITableViewDatasource conformance in Swift
The Swift realm is gonna make things muuuuch more elegant when initializing multiple entity kinds (and hopefully things will be more compact).

Ref. #507 
cc @bummytime (Thanks in advance!!)

### Test
1. Log into your account
2. Verify your notes list loads properly
3. Verify the filters (Tags / Trash) work as expected
4. Verify that swiping over any row displays actions onscreen

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/Classes/SPNoteListViewController+Extensions.swift [hunk 1]
```diff
@@ -131,6 +131,71 @@ extension SPNoteListViewController: UIViewControllerPreviewingDelegate {
 }
 
 
+// MARK: - UITableViewDataSource
+//
+extension SPNoteListViewController: UITableViewDataSource {
```
Comment (bummytime) [inline Simplenote/Classes/SPNoteListViewController+Extensions.swift]: Yay Swift! ❤️

File: Simplenote/Classes/SPNoteListViewController+Extensions.swift [hunk 2]
```diff
@@ -131,6 +131,71 @@ extension SPNoteListViewController: UIViewControllerPreviewingDelegate {
...
+    @objc(configureCell:atIndexPath:)
+    func configure(cell: SPNoteTableViewCell, at indexPath: IndexPath) {
+        let note = fetchedResultsController.object(at: indexPath)
+
+        note.ensurePreviewStringsAreAvailable()
+
+        cell.accessibilityLabel = note.titlePreview
+        cell.accessibilityHint = NSLocalizedString("Open note", comment: "Select a note to view in the note editor")
+
+        cell.accessoryLeftImage = note.published ? .image(name: .shared) : nil
```
Comment (bummytime) [inline Simplenote/Classes/SPNoteListViewController+Extensions.swift]: Thanks for fixing this @jleandroperez !

Comment (jleandroperez) [reply Simplenote/Classes/SPNoteListViewController+Extensions.swift]: Thank **you** Bummy!!

**Comment by jleandroperez** [comment]
Tracking the Autolayout issue, added a note in #507. Will dig deep there when we get to update the NoteTableViewCell (should hopefully be a simple fix).

Thank you @bummytime !!!!

**Pull request merged**